### ISSUE TITLE
Fedora official repository

### ISSUE THREAD

**Issue by tim77** [zerotier/ZeroTierOne#807]
Have you considered push ZeroTier package to official Fedora repo?

How to [Join the package collection maintainers](https://fedoraproject.org/wiki/Join_the_package_collection_maintainers).

**Comment by adamierymenko** [comment]
We've tried. We've also tried to get it into Debian. There are numerous problems.

The #1 issue is that we're a small company and don't have the resources to jump through the many hoops required to babysit official repo packages. We don't have all the time in the world. A secondary and related issue is it means package releases for those distros will lag official releases.

Another major issue is the insistence by distributions on not packaging dependencies. This bit us hard when we tried last time. Two dependencies in particular are very problematic. One is the HTTP parser, which is a minimal C http parser and is the same one used in NodeJS. Its API is not stable, so if we depend on distro packages everything breaks when it's updated. It's a library but it's not really intended to be packaged separately and nodeJS always builds it in. The second one is libminiupnpc. The version in many distros has security vulnerabilities but can't be upgraded because the newer ones have breaking API changes that break other apps, and we are not going to use a vulnerable library even if the vulnerabilities are 'minor'. ZeroTier is used in a lot of really high security environments.

We absolutely loathe dependencies in general and try to avoid them whenever possible, but so far we haven't had the time to develop our own upnpc or http parser libraries. In some cases where other library licenses permit we've actually cut/pasted library code for small libraries (LZ4) into ZeroTier itself to get rid of the dependency, but some in distribution packaging teams noticed this and didn't like it.

This is really a much broader problem than our use case. Present dependency management systems are hopelessly behind the current pace and complexity of software development. I'm not aware of a single dependency management system anywhere that gets it right. This is why you see the popularity of Docker (which treats tar'ed up Linux distributions like binaries) and Go's practice of statically linking everything except libc. Dependencies are an area of CS/IT that desperately needs serious work.
